ballista/executor/src/flight_service.rs

```diff
@@ -105,8 +105,8 @@ impl FlightService for BallistaFlightService {
                 let (tx, rx) = channel(2);
 
                 let executor_id = self.executor_id.clone();
-                task::spawn(async move {
-                    if let Err(e) = read_partition(reader, tx).await {
+                task::spawn_blocking(move || {
```

Comment:
This potentially increases the number of threads by quite a bit.

Reply:
Surprising that there is no async record batch reader

Reply:
time to use `tokio-uring` :)